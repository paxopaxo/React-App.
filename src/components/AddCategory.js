import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('') // Ponerlo como string vacÃ­o para que no vaya undefined en el return
    
    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if ( inputValue.trim().length > 2 ) {

            setCategories( cats => [ inputValue, ...cats ] )
            setInputValue('')
            console.log( inputValue )
        }

    }

    return (
        <form onSubmit = { handleSubmit }>
            <input
                type  = "text"
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}