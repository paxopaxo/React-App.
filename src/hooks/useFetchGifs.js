import { useState, useEffect } from 'react';
import { getGif } from "../helpers/getGif"

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect( () => {
        getGif( category )
            .then( imgs => {

                setTimeout( () => {
                    console.log('aburto')

                    setState({
                        data: imgs,
                        loading: false
                    } )

                }, 3000 )
            })

    }, [category])


    return state 
}