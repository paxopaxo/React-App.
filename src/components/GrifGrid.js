import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

import PropTypes from 'prop-types'

export const GrifGrid = ({ category }) => {

    const { loading, data:images } = useFetchGifs( category )    

    return (
        <>
            <h3> { category }</h3>
            { loading && <p>Cargando.... </p> }
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            // img={ img }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    )
}
GrifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
