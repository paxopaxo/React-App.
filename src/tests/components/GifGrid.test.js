import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import React, { PureComponent } from 'react'
import { GrifGrid } from '../../components/GrifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el <GifGrid />', () => {

    const category = 'One Punch'

    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GrifGrid category= { category }/> )
        expect( wrapper ).toMatchSnapshot()

    })

    test('debe mostrar items cuando se cargan las imagenes useFetch', () => {

        const gifs = [{
            id: 'abdc',
            url:'https://localhost/caca.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GrifGrid  category={ category } /> )

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe( false )
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
        
    })
    



})