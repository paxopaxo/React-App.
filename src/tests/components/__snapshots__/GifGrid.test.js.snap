// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe mostrar items cuando se cargan las imagenes useFetch 1`] = `
<Fragment>
  <h3>
     
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abdc"
      key="abdc"
      title="Cualquier cosa"
      url="https://localhost/caca.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
     
    One Punch
  </h3>
  <p>
    Cargando.... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
